test/db/index.js

```diff
@@ -94,6 +96,164 @@ describe('db', function() {
+      var clientIdA;
+      var clientIdB;
+      var userId;
+      var email;
+
+
+      function seedTokens (client, userId, email, count, expiresAt) {
+        var accessTokens = [];
+        for (var i = 0; i < count; i++) {
+          accessTokens.push({
+            clientId: buf(client),
+            userId: buf(userId),
+            email: email,
+            scope: [auth.SCOPE_CLIENT_MANAGEMENT],
+            expiresAt: expiresAt
+          });
+        }
+
+        return Promise.each(accessTokens, function (options) {
+          return db.generateAccessToken(options);
+        });
+      }
+
+      // Inserts 2000 access tokens with the following breakdown
+      // ClientIdA - 500 expired, 500 valid
+      // ClientIdB - 500 expired, 500 valid
+      before('setup clients', function(){
+        email = 'asdf@asdf.com';
+        clientIdA = randomString(8);
+        clientIdB = randomString(8);
+        userId = buf(randomString(16));
+
+        return db.registerClient({
+            id: clientIdA,
+            name: 'ClientA',
+            hashedSecret: randomString(32),
+            imageUri: 'https://example.domain/logo',
+            redirectUri: 'https://example.domain/return?foo=bar',
+            trusted: true
+          })
+          .then( function () {
+            return db.registerClient({
+              id: clientIdB,
+              name: 'ClientB',
+              hashedSecret: randomString(32),
+              imageUri: 'https://example.domain/logo',
+              redirectUri: 'https://example.domain/return?foo=bar',
+              trusted: true
+            });
+          });
+      });
+
+      beforeEach('seed with tokens', function () {
+        return db._write('DELETE FROM tokens;')
+          .then( function () {
+            return seedTokens(clientIdA, userId, email, 500);
+          })
+          .then( function () {
+            return seedTokens(clientIdB, userId, email, 500);
+          })
+          .then( function () {
+            return seedTokens(clientIdA, userId, email, 500, new Date(Date.now() - (1000 * 600)));
+          })
+          .then( function () {
+            return seedTokens(clientIdB, userId, email, 500, new Date(Date.now() - (1000 * 600)));
+          });
+      });
+
+      it('should fail purgeExpiredTokens without ignoreClientId', function() {
+        return db.purgeExpiredTokens(1000, 5)
+          .then( function () {
+            assert.fail('Purge token should fail with no ignoreClientId');
+          })
+          .catch( function (error) {
+            assert.equal(error.message, 'Invalid ignoreClientId, please ensure client exists.');
+          });
+      });
+
+      it('should call purgeExpiredTokens and ignore client', function() {
+        return db.purgeExpiredTokens(1000, 0, clientIdA)
+          .then( function () {
+            // Check clientA tokens not deleted
+            return db._read('SELECT COUNT(*) FROM fxa_oauth.tokens WHERE clientId=UNHEX(?);', [
```

Comment:
nit: if you made this `SELECT COUNT(*) AS count ..` then the part where you read the result might be slightly cleaner, `result[0].count` rather than `result[0]['COUNT(*)']`.